readthedocs/builds/models.py

```diff
@@ -690,3 +690,128 @@ def run_time(self):
         if self.start_time is not None and self.end_time is not None:
             diff = self.end_time - self.start_time
             return diff.seconds
+
+
+# class VersionAutomationRuleQuerySet(SubclassMixin, models.QuerySet):
+#
+#     """Returns a subclass of VersionAutomationRule, based on the rule type."""
+#
+#     subclass_field = 'rule_type'
+#     subclass_field_id = 'rule_type_id'
+
+
+class VersionAutomationRule(TimeStampedModel, models.Model):
+
+    """Versions automation rules for projects."""
+
+    REGEX_RULE = 'regex-rule'
+    RULE_TYPES = (
+        (REGEX_RULE, _('Regular expression rule')),
+    )
+
+    ACTIVATE_VERSION_ACTION = 'activate-version'
+    ACTIONS = (
+        (ACTIVATE_VERSION_ACTION, _('Activate version on match')),
+    )
+
+    project = models.ForeignKey(
+        Project,
+        related_name='automation_rules',
+        on_delete=models.CASCADE,
+    )
+    priority = models.IntegerField(_('Rule priority'))
+    rule_type = models.CharField(
+        _('Rule type'),
+        max_length=32,
+        choices=RULE_TYPES,
+    )
+    rule_arg = models.CharField(
+        _('Value used for the rule to match the version'),
+        max_length=255,
+    )
+    action = models.CharField(
+        _('Action'),
+        max_length=32,
+        choices=ACTIONS,
+    )
+    action_arg = models.CharField(
```

Comment:
Same here, but I can think of some cases where we could need more than one. But, this also could be solved by creating several rules with different action_arg.

Reply:
How is this `action_arg` used from a user perspective and what is it useful for?

Reply:
Any extra option needed to execute the action. Like, I want to activate the version and set the privacy to public/private. But I can think this can be solved by creating two rules. First activate version `v` and a second one set version `v` to public/private.

Reply:
I'd say that we should keep this simple for now.

If 2 actions are needed for a specific version (match rule), two entries should be created. I'd remove the `action_arg` for now, and add it later if we find that we need it.